kairon/shared/channels/mail/processor.py

```
@@ -0,0 +1,260 @@
+            logger.info(chat_responses)
+
+            for index, response in enumerate(chat_responses):
+                responses[index]['body'] = mp.process_mail(intents[index], response)
+
+            mp.login_smtp()
+            tasks = [mp.send_mail(**response) for response in responses]
+            await asyncio.gather(*tasks)
+            mp.logout_smtp()
+
+        except Exception as e:
+            raise AppException(str(e))
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)